server/msteams/client.go

```diff
@@ -1428,37 +1476,89 @@ func (tc *ClientImpl) SetChatReaction(chatID, messageID, userID, emoji string) e
+	if err != nil {
+		return nil, NormalizeGraphAPIError(err)
+	}
+
+	getMessageRequest, err := tc.client.Chats().ByChatId(chatID).Messages().ByChatMessageId(messageID).ToGetRequestInformation(tc.ctx, nil)
+	if err != nil {
+		return nil, NormalizeGraphAPIError(err)
+	}
+
+	batchRequest := msgraphcore.NewBatchRequest(tc.client.GetAdapter())
+	setReactionRequestItem, err := batchRequest.AddBatchRequestStep(*setReactionRequest)
+	if err != nil {
+		return nil, NormalizeGraphAPIError(err)
+	}
+
+	getMessageRequestItem, err := batchRequest.AddBatchRequestStep(*getMessageRequest)
+	if err != nil {
+		return nil, NormalizeGraphAPIError(err)
+	}
+	getMessageRequestItem.DependsOnItem(setReactionRequestItem)
+
+	return tc.SendBatchRequestAndGetMessage(batchRequest, getMessageRequestItem)
 }
 
-func (tc *ClientImpl) SetReaction(teamID, channelID, parentID, messageID, userID, emoji string) error {
+func (tc *ClientImpl) SetReaction(teamID, channelID, parentID, messageID, userID, emoji string) (*Message, error) {
 	userInfo := map[string]any{
 		"user": map[string]string{
 			"id": userID,
 		},
 	}
 
+	var setReactionRequest *abstractions.RequestInformation
+	var err error
 	if parentID == "" {
 		setReaction := teams.NewItemChannelsItemMessagesItemSetReactionPostRequestBody()
 		setReaction.SetReactionType(&emoji)
 		setReaction.SetAdditionalData(userInfo)
 
-		if err := tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(messageID).SetReaction().Post(tc.ctx, setReaction, nil); err != nil {
-			return NormalizeGraphAPIError(err)
+		setReactionRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(messageID).SetReaction().ToPostRequestInformation(tc.ctx, setReaction, nil)
+		if err != nil {
+			return nil, NormalizeGraphAPIError(err)
 		}
 	} else {
 		setReaction := teams.NewItemChannelsItemMessagesItemRepliesItemSetReactionPostRequestBody()
 		setReaction.SetReactionType(&emoji)
 		setReaction.SetAdditionalData(userInfo)
 
-		if err := tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).SetReaction().Post(tc.ctx, setReaction, nil); err != nil {
-			return NormalizeGraphAPIError(err)
+		setReactionRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).SetReaction().ToPostRequestInformation(tc.ctx, setReaction, nil)
+		if err != nil {
+			return nil, NormalizeGraphAPIError(err)
 		}
 	}
-	return nil
+
+	var getMessageRequest *abstractions.RequestInformation
+	if parentID != "" {
+		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(messageID).Replies().ByChatMessageId1(parentID).ToGetRequestInformation(tc.ctx, nil)
```

Comment:
```suggestion
		getMessageRequest, err = tc.client.Teams().ByTeamId(teamID).Channels().ByChannelId(channelID).Messages().ByChatMessageId(parentID).Replies().ByChatMessageId1(messageID).ToGetRequestInformation(tc.ctx, nil)
```
Check this everywhere.